Use `window-deletable-p' to check if window is deletable.

egh opened the pull request:
Fixes a problem with using `one-window-p' and side windows. The last
non-side window cannot be deleted but `one-window-p' will return `nil'
when there is a side window and a non-side window.

See https://github.com/Alexander-Miller/treemacs/issues/460

ikazuhiro commented:
Please check availability of `window-deletable-p` function.

```lisp
(if (fboundp 'window-deletable-p)
    (window-deletable-p window)
  (one-window-p))
```

It is not directly related with this issue, `one-window-p` usage in `wl-summary.el` looks quite strange.  `(one-window-p fld-win)`  doesn't check whether `fld-win` is only window in a frame.  I guess the below code or similar is expected. 

```lisp
(let ((current-window (current-window)))
  (select-window fld-win)
  (prog1 (one-window-p)
    (select-window current-window)))
```

egh commented:
Thank you! I added a function `wl-window-deletable-p` to wrap it. I also removed the change to the usage in `wl-summary.el` because, as you say, the usage does not seem correct. I can fix that if you would like.

ikazuhiro commented:
> I can fix that if you would like.

Please go on.

egh commented:
I'm looking into this, but I am not sure I understand what is going on. I made the code look like https://github.com/egh/wanderlust/blob/prefer-window-deleteable/wl/wl-summary.el#L3786 which has the same comment and looks like it is supposed to do the same thing. I tested this and it seems to work for me, but I'm not sure I'm doing the right thing.

ikazuhiro commented:
`wl-summary-toggle-disp-msg` had had the similar `delete-window` usage of `wl-summary-toggle-disp-folder`.  It had been updated to the current version at [Feb, 2001](https://github.com/wanderlust/wanderlust/commit/b8853e8bbac5da4cc3e5515858990aad61687854#diff-ef01351fed391538a91e88e9d569c5e5).  But I couldn't find the reason of update.  `one-winodw-p` in `wl-summary-toggle-disp-msg` should always return `nil` because there should be Summary buffer and Folder buffer in a frame.
So I think we can try reverting the ancient change as you do.  Thank you!!

egh commented:
Thank you!
